src/main/resources/templates/java/test/testutils/ClassNameScanner.java

```diff
@@ -228,6 +228,10 @@ private void walkProjectFileStructure(String assignmentFolderName, File node, Ma
                 packageName.lastIndexOf(File.separator + className));
             packageName = packageName.replace(File.separatorChar, '.');
 
+            if (packageName.charAt(0) == '.') {
```

Comment:
Can you add a comment (possibly with some example, which special case is solved here?